xchange-stream-coinmate/src/main/java/info/bitrich/xchangestream/coinmate/CoinmateStreamingAccountService.java

```diff
@@ -67,16 +66,15 @@ public CoinmateStreamingAccountService(PusherStreamingService service, String us
   }
 
   private Observable<Map<String, CoinmateWebsocketBalance>> getCoinmateBalances() {
-    String channelName = "private-user_balances-" + userId;
+    String channelName = "channel/my-balances";
+
+    ObjectReader reader = StreamingObjectMapperHelper.getObjectMapper().readerFor(new TypeReference<Map<String, CoinmateWebsocketBalance>>() {});
+    CoinmateStreamingService service = serviceFactory.createAndConnect(channelName, true);
 
-    return service
-        .subscribeChannel(channelName, "user_balances")
+    return service.subscribeMessages()
         .map(
             (message) -> {
-              Map<String, CoinmateWebsocketBalance> balanceMap =
-                  StreamingObjectMapperHelper.getObjectMapper()
-                      .readValue(
-                          message, new TypeReference<Map<String, CoinmateWebsocketBalance>>() {});
+              Map<String, CoinmateWebsocketBalance> balanceMap = reader.readValue(message.get("balances"));
```

Comment:
nit: can collapse all this down to an expression lambda

Reply:
Thanks, fixed